local/src/core/http.cc

```diff
@@ -1238,7 +1239,9 @@ Session::do_connect(swoc::IPEndpoint const *real_target)
     l.l_onoff = 0;
     l.l_linger = 0;
     setsockopt(socket_fd, SOL_SOCKET, SO_LINGER, (char *)&l, sizeof(l));
-    if (setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &ONE, sizeof(int)) < 0) {
+    if (device.length() && setsockopt(socket_fd, SOL_SOCKET, SO_BINDTODEVICE, device.data(), device.length())) {
```

Comment:
A couple small things:

* Let's use `!device.empty()`. That more explicitly expresses the desired semantic.
* Please compare setsockopt with `< 0` for the error check.

Related to both of these, my general convention in conditions is to explicitly compare against a value in all but boolean function returns or variables.  Thus this is fine:

`if (!device.empty())`

That returns a boolean so a direct condition against that is fine.

However, for all conditions that are not boolean, explicitly compare. Thus this:

`if (ptr != nullptr)`

Not this:

`if (ptr)`

And this:

`if (retval > 0)`

Not this:

`if (retval)`

Let me know if I'm not explaining this well. In doing these explicit comparisons, I'm implementing what is recommended in "Code Complete" chapter 19.1, "Boolean Expressions".